src/tools/ScenarioMeasurement/Startup/TraceSessionManager.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace ScenarioMeasurement
+{
+
+    public interface ITraceSession : IDisposable
+    {
+        void EnableProviders(IParser parser);
+        void EnableKernelProvider(params TraceSessionManager.KernelKeyword[] keywords);
+        void EnableUserProvider(params TraceSessionManager.ClrKeyword[] keywords);
+        void EnableUserProvider(string provider);
+        string GetTraceFilePath();
+    }
+
+    public static class TraceSessionManager
+    {
+        public static ITraceSession CreateSession(string sessionName, string traceName, string traceDirectory, Logger logger)
+        {
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                return new WindowsTraceSession(sessionName, traceName, traceDirectory, logger);
+            }
+            else
+            {
+                return new LinuxTraceSession(sessionName, traceName, traceDirectory, logger);
+            }
+        }
+
+        public static ITraceSession CreateProfileSession(string sessionName, string traceName, string traceDirectory, Logger logger)
```

Comment:
>CreateProfileSession [](start = 36, length = 20)

Three questions:
1) Why does this need to throw? Why can't it just print a message? It doesn't seem like we should fail a helix run (or show a user a nasty exception) in this case if we specify something wrong.
2) What is actually stopping this from getting called on Linux so it won't throw? Is the plan that the script will be special cased to pass in `--skip-profile-iteration` on Linux?
3) Do we have this extra method only to be able to special case the platform? What else makes a "profile" session different from a normal session?

Reply:
Great questions!

1. Print does sound better, so it can just ignore the profile iteration without wasting the previous results of normal iterations.
2. Yes using ```--skip-profile-iteration``` is what I intended. If not specified in Linux runs, as mentioned about the program will print the message and exit 0.
3. Yes the profile session is separated just for the special case of Linux, and I think it is good to have checking platforms only in ```TraceSessionManager.cs```.

Reply:
Got it.

I don't know how I feel about this, honestly. I think that having a special case for the profile iteration so we can flag linux is kind of strange. It's also true that some user mode traces don't work on linux (eventsource) right now, but we aren't doing magic about that. It's also hopefully true that at some point we WILL have Linux profiling support, and then we have this strange API that makes no sense. 

I think it's better to just put this in Startup.cs and set `skipProfileIteration=true` if we're on Linux, and not have a second API here. What do you think about that?